lib/adal.js

```diff
@@ -775,7 +776,13 @@ AuthenticationContext.prototype.getResourceForEndpoint = function (endpoint) {
     
     // default resource will be clientid if nothing specified
     // App will use idtoken for calls to itself
-    if (endpoint.indexOf(this.config.redirectUri) > -1 || (endpoint.indexOf('http://') !== 0 && endpoint.indexOf('https://') !== 0)) {
+    // check if it's staring from http or https, needs to match with app host
+    if (endpoint.indexOf('http://') > -1 || endpoint.indexOf('https://') > -1) {
+        if (this._getHostFromUri(endpoint) === this._getHostFromUri(this.config.redirectUri)) {
+            return this.config.loginResource;
+        }
+    }
+    else {
         return this.config.loginResource;
```

Comment:
Under what circumstances do we end up here?  Getting here would indicate that this is not an http(s) url.  What kind of non-http redirect uri's get used in the context of adal.js?